src/ipc/ipc-common/observer.h

```diff
@@ -0,0 +1,107 @@
+/**********************************************************************
+**
+** Copyright (C) 2020 Luxoft Sweden AB
+**
+** This file is part of the FaceLift project
+**
+** Permission is hereby granted, freIPCServiceAdapterBasee of charge, to any person
+** obtaining a copy of this software and associated documentation files
+** (the "Software"), to deal in the Software without restriction,
+** including without limitation the rights to use, copy, modify, merge,
+** publish, distribute, sublicense, and/or sell copies of the Software,
+** and to permit persons to whom the Software is furnished to do so,
+** subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+** NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+** BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+** ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <QVector>
+#include <QObject>
+
+namespace facelift {
+
+class IObserver : public QObject
+{
+    Q_OBJECT
+public:
+    virtual void onReadyChanged(std::shared_ptr<QMetaObject::Connection> connection) = 0;
+};
+
+
+class IsReadyObserver: public QObject
+{
+    Q_OBJECT
+
+    QVector<IObserver *> m_observers{};
+public:
+    IsReadyObserver() {}
+
+    // Set observers
+    void setObservers(const QVector<IObserver *> &observers) {
+        m_observers = observers;
+        for(auto observer: observers){
+            auto connection = std::make_shared<QMetaObject::Connection>();
+            *connection = QObject::connect(this, &IsReadyObserver::readyChanged, observer, [observer, connection](){
+                observer->onReadyChanged( connection );
+            });
+         }
+    }
+
+    // Get observers
+    const QVector<IObserver *> &getObservers() const {
```

Comment:
Why do we use QVector here, is it better by some reason ?
Please, answer previous comments, because it's quite hard to compare and remember than.

Reply:
Using of QVector as recommedation was noticed at previous comment:
"Have you considered using QVector instead from the point of view the place where we would create the vector ?"
I am currently using QVector for uniform use main types.

Reply:
it was a question:
> Have you considered using QVector... ?

But why is it should be used here ?
E.g. Implicit Sharing matters and so on...

>I am currently using QVector for uniform use main types.

so does QVector used from the call context ?

Reply:
On October 16, 2020 we talked with you about possible use QVector instead std::vector.
Before this moment I have not considered using QVector and have not consulted anyone.
In the last commit, I used QVector instead of std::vector to evaluate the work, performance and for estimation using in future.
I suppose that using QVector is more better way than std::vector because it is full compatible with Qt, but possible it have a constraint or weak sides.
Q: "so does QVector used from the call context ?"
A: Yes.